vispy/visuals/polygon.py

```diff
@@ -38,17 +39,40 @@ class PolygonVisual(Visual):
     """
     def __init__(self, pos=None, color='black',
                  border_color=None, border_width=1, **kwargs):
-        super(PolygonVisual, self).__init__(**kwargs)
-
-        self.mesh = MeshVisual()
-        self.border = LineVisual()
+        self._mesh = MeshVisual()
+        self._border = LineVisual()
         self._pos = pos
         self._color = Color(color)
         self._border_width = border_width
         self._border_color = Color(border_color)
+
         self._update()
-        #glopts = kwargs.pop('gl_options', 'translucent')
-        #self.set_gl_options(glopts)
+        CompoundVisual.__init__(self, [self._mesh, self._border], **kwargs)
+        self._mesh.set_gl_state(polygon_offset_fill=True,
```

Comment:
So, in that case, why does the "wrong" method work for `CubeVisual` and `BoxVisual`?

Is it because the lines that are draw in the `BoxVisual` and `CubeVisual` use `MeshVisual` to draw their borders?

But, that makes no real sense to me, because we set `mode=lines` in the `border` of both `MeshVisual` and `BoxVisual`. So.. why does it work?

The docs read:

> While polygon offset can alter the depth value of filled primitives in point and line 
> mode, under no circumstances will polygon offset affect the depth values of
> GL_POINTS, GL_LINES, GL_LINE_STRIP, or GL_LINE_LOOP primitives. If you are 
>trying to render point or line primitives over filled primitives, use polygon offset to 
>push the filled primitives back. (It can't be used to pull the point and line primitives 
>forward.)